wgpu-core/src/binding_model.rs

```diff
@@ -122,6 +130,47 @@ pub struct BindGroup<B: hal::Backend> {
     pub(crate) life_guard: LifeGuard,
     pub(crate) used: TrackerSet,
     pub(crate) dynamic_count: usize,
+    pub(crate) dynamic_binding_info: Vec<BindGroupDynamicBindingData>,
+}
+
+impl<B: hal::Backend> BindGroup<B> {
+    pub(crate) fn validate_dynamic_bindings(&self, offsets: &[wgt::DynamicOffset]) {
```

Comment:
We are trying to migrate from assertions to error codes. Since this is new code, let's make this right from start and return a `Result<(), BindError>`